webanno/webanno
# #1242 - Forward annotation should load existing annotations

**What's in the PR**
- [x] Refactored forward-annotation code, removed several redundancies, streamlined
- [x] Change signature of Selection.selectSpan to accept CAS instead of JCas
- [x] Re-wrote documentation for forward-annotation mode
- [x] When hitting an existing annotation it is sometimes necessary to press a tag key twice, e.g. if on hits a `NOUN` and `NOUN` is the first tag starting with `N`, then I need to press `N` twice to get to the second tag. Expected would be that pressing `N` moves immediately to the next tag.
- [x] Pressing backspace in forward mode should clear the input field and forward and in the process remove the annotation when a tagset is there

**How to test manually**
* Test annotation with and without forward annotation mode
* Check if scrolling works properly when forwarding onto the end of a page
* Check if annotation stops properly when reaching the end of a document
* Check that existing annotation is loaded properly when hitting one in forward mode and stacking is not enabled
* Check that new annotation is created on top of existing one if stacking is enabled
* Check that clearing the feature value and forwarding actually removes the annotation (with and without tagset)

**Automatic testing**

**Documentation**
